Update remaining tooltip implementations to use generics

frenzibyte opened the pull request.

frenzibyte committed:
Update remaining tooltip implementations to use generics

frenzibyte committed:
Explicitly implement tooltips on date drawables to avoid "convert to auto-property" inspections

bdach commented on osu.Game/Beatmaps/Drawables/DifficultyIconTooltip.cs:
```diff
@@ -0,0 +1,121 @@
+
+namespace osu.Game.Beatmaps.Drawables
+{
+    public class DifficultyIconTooltip : VisibilityContainer, ITooltip<DifficultyIconTooltipContent>
```
Probably should be `internal` at least rather than `public`.

peppy replied:
keep in mind this may break dynamic compilation in weird ways.

bdach replied:
I thought internals are safe for dynamic compilation now? I guess it's not a huge deal but it would be nice to not outwardly expose this.

frenzibyte replied:
Is it? https://github.com/ppy/osu-framework/issues/3727 is still open and I don't recall anything happening regarding it.

bdach replied:
https://github.com/ppy/osu-framework/pull/4006 this one happened after.

frenzibyte committed:
Constrain difficulty icon tooltip to `internal` accessibility

frenzibyte committed:
Revert "Constrain difficulty icon tooltip to `internal` accessibility"

This reverts commit 4a590a041ccf6d755f81f8e052ac092d325ded05.

frenzibyte committed:
Constrain difficulty icon tooltip to `internal` accessibility"

This reverts the reverted commit cd356b8eae1500a2f2127058b7686537d1fc4796.

Sorry for the revert-unrevert, rushly pushed without realizing it
doesn't even build.
